Spigot-API-Patches/0314-Expose-Modified-Enchantment-Level-API.patch

```diff
@@ -0,0 +1,60 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Swedz <me@swedz.net>
+Date: Sat, 29 May 2021 00:12:27 -0400
+Subject: [PATCH] Expose Modified Enchantment Level API
+
+
+diff --git a/src/main/java/org/bukkit/enchantments/Enchantment.java b/src/main/java/org/bukkit/enchantments/Enchantment.java
+index 799813410f40589ba1d7d530a47d87f609764705..287653e9e2911fd8a0fbb18dbfaa84f9cdedf98d 100644
+--- a/src/main/java/org/bukkit/enchantments/Enchantment.java
++++ b/src/main/java/org/bukkit/enchantments/Enchantment.java
+@@ -329,7 +329,23 @@ public abstract class Enchantment implements Keyed {
+      */
+     @NotNull
+     public abstract io.papermc.paper.enchantments.EnchantmentRarity getRarity();
+-
++    
++    /**
++     * Gets the modified enchantment level minimum for the enchantment given the level
++     *
++     * @param level the enchantment level
++     * @return the minimum modified enchantment level
++     */
++    public abstract int getModifiedEnchantmentLevelMinimum(int level);
++    
++    /**
++     * Gets the modified enchantment level maximum for the enchantment given the level
++     *
++     * @param level the enchantment level
++     * @return the maximum modified enchantment level
++     */
++    public abstract int getModifiedEnchantmentLevelMaximum(int level);
++    
+     /**
+      * Gets the damage increase as a result of the level and entity category specified
+      *
+diff --git a/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java b/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java
+index ca097db904db96e81c0ea4b06b445a13e9333cfe..f0763d4ddc54531ca1387a8b23338bee7c6b4f66 100644
+--- a/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java
++++ b/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java
+@@ -85,7 +85,19 @@ public class EnchantmentWrapper extends Enchantment {
+     public io.papermc.paper.enchantments.EnchantmentRarity getRarity() {
+         return getEnchantment().getRarity();
+     }
+-
++    
++    @Override
++    public int getModifiedEnchantmentLevelMinimum(int level)
++    {
```

Comment:
idk how style-nazi paper is, but like everywhere else the opening bracket is on the same line as the function definition

Reply:
Yeah that's my bad. My IntelliJ is configured to work like that so it just automatically did it like that and I thought nothing of it.

Reply:
Also, make sure to remove all whitespace on empty lines, if you don't, everyone elses IDEs will remove it automatically and itll just create diff anoyingness for others trying to modify the file without adding those extra whitespace removal changes.

Reply:
Sure thing. How can I change it? Do I need to make another commit on my fork? Sorry I'm new to PRs, trying to figure this all out.

Reply:
So you make the changes in the API/Server folders, then just amend the previous commit to include those changes
* `git add .`
* `git commit --amend`
* rebuild patches
* commit changes to the main repo
* push to your remote branch